libtenzir/src/execution_node.cpp

```diff
@@ -93,6 +94,32 @@ auto make_timer_guard(Duration&... elapsed) {
     });
 }
 
+// Return an underestimate for the total number of referenced bytes for a vector
+// of table slices, excluding the schema and disregarding any overlap or custom
+// information from extension types.
+auto num_approx_bytes(const std::vector<table_slice>& events) {
```

Comment:
Note that the result of this function can be off by a large margin. It never overestimates, but sometimes the result is a lot smaller than you would think and also a lot smaller than it should be. 

That happens because it is not trivial for Arrow to get the referenced byte ranges from all underlying array data buffers. Sometimes, getting that information accurately is impossible.

However, I opted to use the built-in Arrow solution here hoping that it will be improved upon in the future upsrream, rather than us having to roll our own.